Library/Homebrew/cmd/info.rb

```diff
@@ -61,25 +61,29 @@ def info_args
 
   def info
     info_args.parse
+
     if args.days.present?
-      raise UsageError, "days must be one of #{VALID_DAYS.join(", ")}" unless VALID_DAYS.include?(args.days)
+      raise UsageError, "--days must be one of #{VALID_DAYS.join(", ")}" unless VALID_DAYS.include?(args.days)
     end
 
     if args.category.present?
       if ARGV.named.present? && !VALID_FORMULA_CATEGORIES.include?(args.category)
-        raise UsageError, "category must be one of #{VALID_FORMULA_CATEGORIES.join(", ")} when querying formulae"
+        raise UsageError, "--category must be one of #{VALID_FORMULA_CATEGORIES.join(", ")} when querying formulae"
       end
 
       unless VALID_CATEGORIES.include?(args.category)
-        raise UsageError, "category must be one of #{VALID_CATEGORIES.join(", ")}"
+        raise UsageError, "--category must be one of #{VALID_CATEGORIES.join(", ")}"
       end
     end
 
     if args.json
-      raise UsageError, "invalid JSON version: #{args.json}" unless ["v1", true].include? args.json
+      raise UsageError, "Invalid JSON version: #{args.json}" unless ["v1", true].include? args.json
+      raise UsageError, "This command's option requires a formula argument" if ARGV.named.empty?
```

Comment:
I made these a `UsageError` with a custom message stating that a formula argument is required only for these specific flags, rather than the command in general, as is implied in the wording of `FormulaUnspecifiedError`.

Reply:
Gotcha 👍